mbuild/lattice.py

```diff
@@ -0,0 +1,401 @@
+                                 .format(dimension))
+        else:
+            raise ValueError('Lattice Spacing Issue: None provided, '
+                             'must provide lattice spacings matching '
+                             'the dimension ({}) of the system.'
+                             .format(dimension))
+        print(lattice_spacings)
+        if np.any(lattice_spacings <= 0.0):
+            raise ValueError('Negative or zero lattice spacing value. One of '
+                             'the spacings {} is negative or 0.'
+                             .format(lattice_spacings))
+
+        if basis_vectors is None:
+            basis_vectors = defaultdict(list)
+            if dimension == 3:
+                basis_vectors['default'].append((0, 0, 0))
+            elif dimension == 2:
+                basis_vectors['default'].append((0, 0))
+            else:
+                basis_vectors['default'].append((0))
+        elif (isinstance(basis_vectors, list) or
+              isinstance(basis_vectors, tuple)):
+            for lst in basis_vectors:
+                if len(lst) != 2:
+                    raise ValueError('Too many arguments per basis vector. '
+                                     '{} was provided, but the format should '
+                                     'be: vector = ( (\'ID1\' , [x1,y1,z1]), '
+                                     '( \'ID2\' , [x2,y2,z2]) )'.format(lst))
+                elif not isinstance(lst[0], str):
+                    raise TypeError('ID Error, not a string. {} is not of '
+                                    'type str. Please correct in the '
+                                    'basis_vectors input {}.'
+                                    .format(lst[0], basis_vectors))
+
+                if len(lst[1]) == dimension:
+                    if isinstance(lst[1], tuple) or isinstance(lst[1], list):
+                        for i in range(dimension):
+                            if (lst[1][i] < 0 or lst[1][i] >= 1):
+                                raise ValueError('Incorrect basis fractions. '
+                                                 '{} is supposed be 0 >= and '
+                                                 '< 1.'.format(lst[1]))
+
+                    else:
+                        raise TypeError('Incorrect Format: expected <list> '
+                                        'or <tuple>. {} was passed in as '
+                                        ' type: {} from basis_vectors.'
+                                        .format(lst[1], type(lst[1])))
+                else:
+                    raise ValueError('Basis Vector does not match dimension. '
+                                     '{} does not have vector components '
+                                     'equal to the dimension specified: {}'
+                                     .format(lst[1], dimension))
+        else:
+            raise TypeError('Incorrect Type: "basis_vectors" is of type {}, '
```

Comment:
Where is the case for the user passing the already correct defaultdict? I think you might just want to enforce users providing a dict and not even allowing lists/tuples. Whenever it doesn't limit us, provide one and only one correct way to do something.